topic_dynamics/parsing.py

```diff
@@ -0,0 +1,427 @@
+                           .format(date=date, start_index=str(diff_indices[date][0]),
+                                   end_index=str(diff_indices[date][1])))
+
+
+def uci_format(tokens_file: str, output_dir: str, name: str) -> None:
+    """
+    Transform the file with tokens into the UCI bag-of-words format. The format consists of two files:
+    the first one lists all the tokens in the dataset alphabetically, and the second one lists all the
+    triplets document-token-count, ranged first by documents, then by tokens.
+    :param tokens_file: the path to the temporary file with tokens.
+    :param output_dir: path to the output directory.
+    :param name: name of the processed dataset.
+    :return: None.
+    """
+    print("Transforming the data into the UCI format for topic-modeling.")
+    n_nnz = 0
+    set_of_tokens = set()
+    # Compile a list of all tokens in the dataset for a sorted list
+    with open(tokens_file) as fin:
+        for n_documents, line in enumerate(fin, start=1):
+            token_line = parse_token_line(line)
+            for token in token_line.tokens.split(","):
+                n_nnz = n_nnz + 1
+                set_of_tokens.add(token.split(":")[0])
+    n_tokens = len(set_of_tokens)
+    # Sort the list of tokens, transform them to indexes and write to file
+    sorted_list_of_tokens = sorted(list(set_of_tokens))
+    sorted_dictionary_of_tokens = {}
+    with open(os.path.abspath(os.path.join(output_dir, "vocab." + name + ".txt")), "w+") as fout:
+        for index in range(len(sorted_list_of_tokens)):
+            sorted_dictionary_of_tokens[sorted_list_of_tokens[index]] = index + 1
+            fout.write(sorted_list_of_tokens[index] + "\n")
+    # Compile the second necessary file: NNZ triplets sorted by document
+    with open(tokens_file) as fin, open(
+            os.path.abspath(os.path.join(output_dir, "docword." + name + ".txt")), "w+") as fout:
+        fout.write(str(n_documents) + "\n" + str(n_tokens) + "\n" + str(n_nnz) + "\n")
+        for line in tqdm(fin):
+            token_line = parse_token_line(line)
+            file_tokens = token_line.tokens.split(",")
+            file_tokens_separated = []
+            file_tokens_separated_numbered = []
+            for entry in file_tokens:
+                file_tokens_separated.append(entry.split(":"))
+            for entry in file_tokens_separated:
+                file_tokens_separated_numbered.append([sorted_dictionary_of_tokens[entry[0]], int(entry[1])])
+            file_tokens_separated_numbered = sorted(file_tokens_separated_numbered, key=itemgetter(0), reverse=False)
+            for entry in file_tokens_separated_numbered:
+                fout.write("{doc_id} {token_id} {count}\n".format(doc_id=str(line.split(";")[0]),
+                                                                  token_id=str(entry[0]), count=str(entry[1])))
+
+
+def slice_and_parse_full_files(repository: str, output_dir: str, n_dates: int,
```

Comment:
what does it mean `full file`?

Reply:
It's the opposite of diffs. It means that diffs won't be calculated and that the UCI format (for topic modeling) will be created for the identifiers of the whole files. Probably, we personally won't use it much, since diffs are better, but since it was already written, I left this functionality. The name seems understandable enough for me!